viewing_party/party.py

```diff
@@ -1,23 +1,163 @@
+        genre_count = genre_dict[genre]
+        if genre_count > most_freq_genre[1]:
+            most_freq_genre = (genre, genre_count)
+
+    return most_freq_genre[0]
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+    watched_lst = user_data["watched"]
+    friends_lst = user_data["friends"]
+    unique_watched_lst = []
+
+    for movie in watched_lst:
+        if not one_friend_has_seen(movie, friends_lst):
+            unique_watched_lst.append(movie)
+
+    return unique_watched_lst
+
+    # Another Solution
+    # for movie in watched_lst:
+    #     seen_movie = False
+    #     for friend in friends_lst:
+    #         if movie in friend["watched"]:
+    #             seen_movie = True
+        
+    #     if seen_movie == False:
+    #         unique_watched_lst.append(movie)
+
+
+def get_friends_unique_watched(user_data):
+    user_watched_lst = user_data["watched"]
+    friends_lst = user_data["friends"]
+
+    user_not_watched = []
+
+    for friend in friends_lst:
+        for movie in friend["watched"]:
+            if movie not in user_watched_lst and movie not in user_not_watched:
+                user_not_watched.append(movie)
+
+    return user_not_watched
+
+
+#  Helper to check if movie watched by at least one friend 
+def one_friend_has_seen(movie, friendList): 
+    for friend in friendList:
+        if movie in friend["watched"]:
+            return True
+    return False
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    user_watched_lst = user_data["watched"]
+    friends_lst = user_data["friends"]
+    subscriptions = user_data["subscriptions"]
+    
+    recommended_movies = []
+
+    for friend in friends_lst:
+        for movie in friend["watched"]:
+            if movie not in user_watched_lst and movie["host"] in subscriptions and movie not in recommended_movies:
+                recommended_movies.append(movie)
+
+    return recommended_movies
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    user_watched_lst = user_data["watched"]
+    friends_lst = user_data["friends"]
+
+    recommended_movies = []
+    most_freq_genre = get_most_watched_genre(user_data)
+
+    for friend in friends_lst:
+        for movie in friend["watched"]:
+            if movie not in user_watched_lst and movie["genre"] == most_freq_genre and movie not in recommended_movies:
+                recommended_movies.append(movie)
+
+    return recommended_movies
+
+
+def get_rec_from_favorites(user_data):
+    friends_lst = user_data["friends"]
+    users_favorites = user_data["favorites"]
+
+    rec_from_favorites = list(users_favorites)

```

Comment:
Using `list()` usually implies converting a variable to a list. Seems you're using it to make a copy of user_data["favorites"] which is already a list? Good idea- it will help avoid side effects. Perhaps `.copy()` might be more easily understood your intention behind the code!